### PR TITLE
Defines the API for the static maps

### PR DESCRIPTION
**Pull request by javierarce**
This branch adds a new object (`cdb.Image`) defined in [src/core/image.js](https://github.com/CartoDB/cartodb.js/blob/259-medusa/src/core/image.js) that uses the new Static Api to generate images from visualizations.

## Changed/added files:

- [src/cartodb.js](https://github.com/CartoDB/cartodb.js/blob/259-medusa/src/cartodb.js)
- [src/core/image.js](https://github.com/CartoDB/cartodb.js/blob/259-medusa/src/core/image.js)
- [core/image.spec.js](https://github.com/CartoDB/cartodb.js/blob/259-medusa/test/spec/core/image.spec.js)
- [examples/image.html](https://github.com/CartoDB/cartodb.js/blob/259-medusa/examples/image.html)

### CODE REVIEW

File: src/core/image.js [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+
+    this.supported_formats = ["png", "jpg"];
+
+    this.defaults = {
+      tiler_domain: "cartodb.com",
+      tiler_port: "80"
+    };
+
+    this.available_basemaps = ["light_all", "light_nolabels", "dark_all", "dark_nolabels"];
+    this.options = _.defaults({
```
Comment (javisantana) [inline src/core/image.js]: this constructor is almost a copy for Map constructor, shouldn't we call Map constructor instead?

File: src/core/image.js [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+
+      var ld = {
+        layers: layers
+      };
+
+      return ld;
+
+    },
+
+    _requestPOST: function(params, callback) {
```
Comment (javisantana) [inline src/core/image.js]: this should not be reimplemented here

File: src/core/image.js [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+          callback(data);
+        },
+        error: function(xhr) {
+          callback(null);
+        }
+      });
+
+    },
+
+    _tilerHost: function() {
```
Comment (javisantana) [inline src/core/image.js]: should use Map one

File: src/core/image.js [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+      var opts = this.options;
+      return opts.tiler_protocol +
+        "://" + ((opts.user_name) ? opts.user_name+".":"")  +
+      opts.tiler_domain +
+        ((opts.tiler_port != "") ? (":" + opts.tiler_port) : "");
+    },
+
+    _host: function(subhost) {
```
Comment (javisantana) [inline src/core/image.js]: should use Map one

File: src/core/image.js [hunk 5]
```diff
@@ -0,0 +1,539 @@
...
+      var file = "cartodb.mod." + modName + (cartodb.DEBUG ? ".uncompressed.js" : ".js");
+      var src = this.getPath(file);
+      if (!src) {
+        cartodb.log.error("can't find cartodb.js file");
+      }
+      ImageLoader.loadScript(src);
+    }
+  };
+
+  window.imgjson = function(data) {
```
Comment (javisantana) [inline src/core/image.js]: this should not be duplicated from vizjson loader

File: src/core/image.js (hunk 3 above)
Comment (javierarce) [inline src/core/image.js]: :ok_hand: removed

File: src/core/image.js (hunk 2 above)
Comment (javierarce) [inline src/core/image.js]: :ok_hand: removed

File: src/core/image.js (hunk 1 above)
Comment (javierarce) [inline src/core/image.js]: :ok_hand: I changed the constructor to use the Map one.

File: src/core/image.js (hunk 4 above)
Comment (javierarce) [inline src/core/image.js]: :ok_hand: removed

File: src/core/image.js (hunk 5 above)
Comment (javierarce) [inline src/core/image.js]: :ok_hand: we're now using the vizjson loader

**Comment by javisantana** [comment]
:es:

**Pull request merged**